src/Fakta/Status.fs

```diff
@@ -1,9 +1,44 @@
 ﻿module Fakta.Status
+open Fakta
+open Fakta.Logging
+open Fakta.Impl
+open System
+open System.Collections
+open NodaTime
+open HttpFs.Client
+open Chiron
+
+
+let faktaStatusString = "Fakta.status"
+
+let statusDottedPath (funcName: string) =
+  (sprintf "%s.%s" faktaStatusString funcName)
+
 
 /// Leader is used to query for a known leader 
-let leader (s : FaktaState) : Async<Choice<string, Error>> =
-  raise (TBD "TODO")
+let leader (state : FaktaState) : Async<Choice<string, Error>> =  
+    let urlPath = "leader"
+    let uriBuilder = UriBuilder.ofStatus state.config urlPath
+    let result = Async.RunSynchronously (call state (statusDottedPath urlPath) id uriBuilder HttpMethod.Get)
+    async {
+      match result with 
+      | Choice1Of2 x -> 
+          let body, (dur:Duration, resp:Response) = x
+          let item = if body = "" then "" else Json.deserialize (Json.parse body)
```

Comment:
Use `Json.tryDeserialize` and `Json.tryParse` for production code. It's in the `Choice<_,_>` monad, so you should be able to do: `results |> Choice.bind (fun (dur, resp) -> ...` etc.